React/Base/RCTComponentEvent.h

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <React/RCTEventDispatcher.h>
+
+/**
+ * Event that is dispatched to React components.
+ */
+@interface RCTComponentEvent : NSObject<RCTEvent>
```

Comment:
Should we add a comment here that this type of Event does not have `reactTag`?

Reply:
Should we emphasize somehow in the names that it is untyped, simple, event which is legit but... we discourage to use it because it is untyped?

Reply:
It is a generic untyped event for Components because it DOES use view tags (target that is extracted from the body). We could say that it is preferred to create a custom event class that implements RCTEvent for type safety. The main use case of this class is it's usage internally in RCTDirectEventBlock and RCTBubblingEventBlock. It can also be used to migrate code that used sendInputEvent directly without creating a custom class.

Reply:
Yeah! We have to mention this in comment and probably reflect somehow in the name.